kratos/linear_solvers/amgcl_solver.h

```diff
@@ -543,39 +543,70 @@ class AMGCLSolver : public LinearSolver< TSparseSpaceType,
         ) override
     {
         int old_ndof = -1;
-        unsigned int old_node_id = rDofSet.size() ? rDofSet.begin()->Id() : 0;
         int ndof=0;
-        for (auto it = rDofSet.begin(); it!=rDofSet.end(); it++) {
-            if(it->EquationId() < TSparseSpaceType::Size1(rA) ) {
-                IndexType id = it->Id();
-                if(id != old_node_id) {
-                    old_node_id = id;
-                    if(old_ndof == -1) old_ndof = ndof;
-                    else if(old_ndof != ndof) { //if it is different than the block size is 1
-                        old_ndof = -1;
-                        break;
-                    }
 
-                    ndof=1;
-                } else {
-                    ndof++;
+        if(rModelPart.GetCommunicator().GetDataCommunicator().IsDistributed() == false)
```

Comment:
```suggestion
        if (!rModelPart.IsDistributed())
```

Reply:
Oo Can this be called directly without the communicator?

Reply:
[Yep](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/model_part.h#L1707-L1710)